python/ray/tune/trial.py

```diff
@@ -325,8 +337,16 @@ def _setup_resources(self, log_always: bool = False):
 
 
         """
-        if not self.placement_group_factory and \
-           not int(os.getenv("TUNE_PLACEMENT_GROUP_AUTO_DISABLED", "0")):
+        # Placement groups are force-disabled via env variable.
+        if int(os.getenv("TUNE_PLACEMENT_GROUP_AUTO_DISABLED", "0")):
+            if self.placement_group_factory:
+                self.resources = pg_factory_to_resources(
+                    self.placement_group_factory)
+            self.placement_group_factory = None
+
+        # Placement groups are not disabled, but none is given.
+        # Produce one automatically from self.resources.
+        elif not self.placement_group_factory:
```

Comment:
Ah, this semantically alters the previous behavior, where `TUNE_PLACEMENT_GROUP_AUTO_DISABLED` only disabled *automatic* conversion of resources to placement groups but, importantly, allowed manual opt-in to using placement groups when passed in `resources_per_trial`. See https://docs.ray.io/en/master/tune/user-guide.html#environment-variables

Generally if a user passes a placement group factory or provides a PGF as a default resource request we would still want to use placement groups (otherwise they just shouldn't pass them).

If we want to enable users to use the old-style Resources for RLLib trainables, we should move the conversion of PGF to Resources into the `default_resource_request()` methods of the `Trainer` (and maybe use a different environment variable? That's up to you).

Reply:
Makes sense. Thanks for the catch.